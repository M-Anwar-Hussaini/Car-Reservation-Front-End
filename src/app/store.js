import { combineReducers, configureStore } from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import AppDataSlice from './redux/AppDataSlice';
import carDetailsSlice from './redux/carDetailsSlice';
import { authReducer } from './redux/auth/authSlice';

const rootReducer = combineReducers({
  appData: AppDataSlice,
  carDetails: carDetailsSlice,
  auth: authReducer,
});

const persistConfig = {
  key: 'root',
  whitelist: ['auth'],
  storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }),
});

export const persistor = persistStore(store);

export default store;
