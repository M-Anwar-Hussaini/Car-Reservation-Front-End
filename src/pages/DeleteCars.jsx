import React from 'react';

const DeleteCars = () => (
  <div>
    <h1>Delete Cars</h1>
    <p>List of available cars that can be deleted</p>
  </div>
);

export default DeleteCars;
