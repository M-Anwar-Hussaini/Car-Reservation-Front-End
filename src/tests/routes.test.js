import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { PersistGate } from 'redux-persist/lib/integration/react';
import store, { persistor } from 'app/store';
import Layout from 'pages/Layout';
import Home from 'pages/Home';
import AddCar from 'pages/AddCar';
import Reserve from 'pages/AddReserve';
import CarDetails from 'pages/CarDetails';
import DeleteCars from 'pages/DeleteCars';
import Missing from 'pages/Missing';
import MyReservation from 'pages/MyResevation';
import Register from 'pages/Register';
import Login from 'pages/Login';

describe('Test for routes', () => {
  it('Home page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <Home />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Add car page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <AddCar />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Add reservation page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <Reserve />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Car detail page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <CarDetails />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Delete car page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <DeleteCars />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Missing page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <Missing />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('My reservations page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <MyReservation />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Register page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <Register />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Login page renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <PersistGate loading={null} persistor={persistor}>
            <BrowserRouter>
              <Layout>
                <Login />
              </Layout>
            </BrowserRouter>
          </PersistGate>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
