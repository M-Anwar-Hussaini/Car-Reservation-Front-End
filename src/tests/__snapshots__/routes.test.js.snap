// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for routes Add car page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;

exports[`Test for routes Add reservation page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;

exports[`Test for routes Car detail page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;

exports[`Test for routes Delete car page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;

exports[`Test for routes Home page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  />,
]
`;

exports[`Test for routes Login page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;

exports[`Test for routes Missing page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;

exports[`Test for routes My reservations page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;

exports[`Test for routes Register page renders correctly 1`] = `
Array [
  <div
    className="flex justify-between"
  >
    <div
      className="lg:hidden"
    >
      <button
        className="fixed top-0 left-0 z-50 p-3 focus:outline-none"
        onClick={[Function]}
        type="button"
      >
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
        <div
          className="bg-black h-[2px] w-7 m-1 transform transition-all duration-300 origin-top-left group-focus:translate-y-6"
        />
      </button>
    </div>
    <div
      className="lg:block lg:relative w-2/3 md:w-65 lg:w-64 lg:flex-none bg-white z-40 h-full hidden"
    >
      <aside
        aria-label="Sidebar"
        className="flex flex-col lg:p-auto items-center w-100 lg:w-60 h-full border-r-2"
      >
        <div
          className="pb-12 pt-4 flex flex-col items-center"
        >
          <img
            alt="logo"
            className="w-40"
            src="transparent-logo.png"
          />
          <h1
            className="text-light-green font-bold text-5xl text-center"
          >
            Book
            <span
              className="font-medium"
            >
              Car
            </span>
          </h1>
        </div>
        <nav
          className="w-full pl-8"
        >
          <ul
            className="flex flex-col space-y-2"
          >
            <li>
              <a
                aria-current="page"
                className="w-full inline-block bg-light-green px-4 py-2 font-bold text-white text-lg"
                href="/"
                onClick={[Function]}
              >
                CARS
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations"
                onClick={[Function]}
              >
                MY RESERVATION
              </a>
            </li>
            <li>
              <a
                className="w-full inline-block px-4 py-2 hover:bg-light-green hover:text-white font-bold text-lg"
                href="/reservations/new"
                onClick={[Function]}
              >
                RESERVE A CAR
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="mt-auto"
        >
          <p
            className="text-center pb-4"
          >
            <a
              className="font-bold text-lg"
              href="/login"
              onClick={[Function]}
            >
              LOGIN
            </a>
            <span
              className="text-lg"
            >
               / 
            </span>
            <a
              className="font-bold text-lg"
              href="/register"
              onClick={[Function]}
            >
              REGISTER
            </a>
          </p>
          <div>
            <p
              className="text-center"
            >
              <a
                className="text-dark-blue"
                href="http://localhost:4000/api-docs"
                rel="noreferrer"
                target="_blank"
              >
                API Documentation
              </a>
            </p>
            <p
              className="text-center text-dark-blue"
            >
              © 2024 BookCar
            </p>
            <p
              className="text-center text-dark-blue"
            >
              All rights reserved
            </p>
          </div>
        </div>
      </aside>
    </div>
  </div>,
  <main
    className="flex-1 text-black overflow-auto h-full px-3 py-5 bg-gray-100"
  >
    <div>
      Loading...
    </div>
  </main>,
]
`;
